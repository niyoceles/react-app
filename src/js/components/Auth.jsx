import React, { Component } from 'react';
import { connect } from 'react-redux';
import uuidv1 from 'uuid';
import { loginData, signUp } from '../actions/index';

export class Auth extends Component {
  constructor() {
    super();
    this.state = {
      showModalSignUp: false,
      showModalSignIn: false,
      firstName: '',
      lastName: '',
      email: '',
      phone: '',
      userName: '',
      password: '',
      location: ''
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleSubmitSignup = this.handleSubmitSignup.bind(this);

    this.toggleModalsignUp = this.toggleModalsignUp.bind(this);
    this.toggleModalsignIn = this.toggleModalsignIn.bind(this);
  }

  toggleModalsignUp() {
    const { showModalSignUp } = this.state;
    this.setState({ showModalSignUp: !showModalSignUp });
  }

  toggleModalsignIn() {
    const { showModalSignIn } = this.state;
    this.setState({ showModalSignIn: !showModalSignIn });
  }

  handleChange(event) {
    this.setState({ [event.target.id]: event.target.value });
  }

  handleSubmitSignup(e) {
    e.preventDefault();
    const {
      firstName, lastName, email, phone, userName, password, location
    } = this.state;

    const signUpData = {
      id: uuidv1(),
      firstName,
      lastName,
      email,
      phone,
      userName,
      password,
      location
    };
    // call action
    // eslint-disable-next-line react/prop-types
    this.props.signUp(signUpData);
  }

  handleSubmit(event) {
    event.preventDefault();
    const postLoginData = {
      email: this.state.email,
      password: this.state.password
    };
    this.props.loginData(postLoginData);
    console.log(this);
  }

  render() {
    const { showModalSignUp } = this.state;
    const { showModalSignIn } = this.state;

    // const { email } = this.state;
    // const { password } = this.state;
    // Getting the login data and token
    const userLogin = this.props.aaa.login;
    localStorage.setItem('token', userLogin.token);
    localStorage.setItem('userDetails', JSON.stringify(userLogin.data));
    if (userLogin.status === 200) {
      alert('Successful login');
      window.location = './users';
    }
    if (userLogin.status === 400) {
      alert(userLogin.error);
    }

    const {
      firstName, lastName, email, phone, userName, password, location
    } = this.state;

    const userSignup = this.props.newUser.item;
    localStorage.setItem('token', userSignup.token);
    localStorage.setItem('userDetails', JSON.stringify(userSignup.data));
    if (userSignup.status === 201) {
      alert('Successful signup');
      window.location = './users';
    }

    // eslint-disable-next-line no-nested-ternary
    return showModalSignUp ? (
      <div className="container">
        <div className="navbar">
          <div className="navbar-header">
            <div className="navbar-title">
              <div className="logo">
                <img src={require('../../assets/logo.png')} alt="" />
              </div>
            </div>
          </div>
          <div className="navbar-btn">
            <label htmlFor="navbar-check">
              <span />
              <span />
              <span />
            </label>
          </div>
          <input type="checkbox" id="navbar-check" />
          <div className="navbar-links">
            <a href="./users">User</a>
            {' '}
            <button id="sign-up" onClick={this.toggleModalsignUp}>Sign Up</button>
            <button id="sign-in" onClick={this.toggleModalsignIn}>Sign In</button>
          </div>
        </div>
        <form className="animate" id="signUpForm" onSubmit={this.handleSubmitSignup}>
          <h3> Sign Up</h3>
          <label>First name</label>
          <input
            type="text"
            placeholder="Enter First name"
            id="firstName"
            value={firstName}
            onChange={this.handleChange}
            required
          />
          <div className="error" id="firstname-error" />
          <label>Last name</label>
          <input
            type="text"
            placeholder="Enter Last name"
            id="lastName"
            value={lastName}
            onChange={this.handleChange}
            required
          />
          <div className="error" id="lastname-error" />
          <label>Email</label>
          <input
            type="email"
            placeholder="Enter Email"
            id="email"
            value={email}
            onChange={this.handleChange}
            required
          />
          <div className="error" id="email-error" />
          <label>Phone Number</label>
          <input
            type="number"
            placeholder="Enter Phone number"
            id="phone"
            value={phone}
            onChange={this.handleChange}
            required
          />
          <div className="error" id="phone-error" />
          <label>Username</label>
          <input
            type="text"
            placeholder="Enter Username"
            id="userName"
            value={userName}
            onChange={this.handleChange}
            required
          />
          <div className="error" id="username-error" />
          <label>Password</label>
          <input
            type="password"
            placeholder="Enter Password"
            id="password"
            value={password}
            onChange={this.handleChange}
            required
          />
          <div className="error" id="password-error" />
          <label>Location</label>
          <input
            type="text"
            placeholder="Enter location"
            id="location"
            value={location}
            onChange={this.handleChange}
            required
          />
          <div className="error" id="location-error" />
          <div className="error" id="connection-error" />
          <button type="submit" id="btnSignUp">Sign Up</button>
          <div className="success" id="success-signup" />
          <div className="error" id="account-exist" />
        </form>
        <p className="signInlink">
Are you a Member?
          {' '}
          <button onClick={this.toggleModalsignUp}>Sign In clinke </button>
          {' '}
        </p>
      </div>
    ) : showModalSignIn ? (

      <div className="container">
        <div className="navbar">
          <div className="navbar-header">
            <div className="navbar-title">
              <div className="logo">
                <img src={require('../../assets/logo.png')} alt="" />
              </div>
            </div>
          </div>
          <div className="navbar-btn">
            <label htmlFor="navbar-check">
              <span />
              <span />
              <span />
            </label>
          </div>
          <input type="checkbox" id="navbar-check" />
          <div className="navbar-links">
            <a href="./users">User</a>
            {' '}
            <button id="sign-up" onClick={this.toggleModalsignUp}>Sign Up</button>
            <button id="sign-in" onClick={this.toggleModalsignIn}>Sign In</button>
          </div>
        </div>
        <form className="animate" id="signInForm" onSubmit={this.handleSubmit}>
          <h3> Sign In page</h3>
          <label>Email</label>
          <input
            type="email"
            placeholder="Enter Email"
            name="email"
            id="email"
            value={email}
            onChange={this.handleChange}
            required
          />
          <label>Password</label>
          <input
            type="password"
            placeholder="Enter Password"
            name="password"
            id="password"
            value={password}
            onChange={this.handleChange}
          />
          <div className="error" id="incorrect-error" />
          <div className="error" id="connection-error-login" />
          <button type="submit" id="btn-sign-in" className="btn-submit">Sign In </button>
          <div className="success" id="success-login" />
          <p className="signUplink">
Not yet a Member?
            {' '}
            <button onClick={this.toggleModalsignUp}> Sign Up Link1 </button>
            {' '}
          </p>
        </form>
      </div>
    ) : (
      <div>
        <div className="navbar">
          <div className="navbar-header">
            <div className="navbar-title">
              <div className="logo">
                <img src={require('../../assets/logo.png')} alt="" />
              </div>
            </div>
          </div>
          <div className="navbar-btn">
            <label htmlFor="navbar-check">
              <span />
              <span />
              <span />
            </label>
          </div>
          <input type="checkbox" id="navbar-check" />
          <div className="navbar-links">
            <a href="./users">User</a>
            {' '}
            <button id="sign-up" onClick={this.toggleModalsignUp}>Sign Up</button>
            <button id="sign-in" onClick={this.toggleModalsignIn}>Sign In</button>
          </div>
        </div>
        <div className="container">
          <h1>Welcome to Banka Application </h1>
        </div>

        <section id="main-section">
          <div className="container">
            <div className="panel">
              <img src={require('../../assets/open-account.jpg')} alt="Service" />
              <h1>Opening Account:</h1>
              <h3>Feel free to open an account by signup here</h3>
              <button className="get-stared-link">Get Started</button>
            </div>
            <div className="panel">
              <img src={require('../../assets/loan1.jpg')} alt="security" />
              <h1>Get a loan with us:</h1>
              <h3>
After getting an account,
                <br />
                <br />
                    you can start to request a loan
                <br />
              </h3>

            </div>
            <div className="panel">
              <img src={require('../../assets/security.jpg')} alt="Contact" />
              <h1>Feel the Security of your money</h1>
              <h3>We are here to make safe of your money security and profitability</h3>
            </div>
          </div>
        </section>
        <section id="main-section">
          <div className="container">
            <div className="panel2">
              <h2>Contact us</h2>
              <h3> Do you need any advice or helps?</h3>
              <h3>
Please Feel Free to Contact on our
                <address>Call (250) 783-067-644.</address>
                <address>Mail: celestin@banka.com</address>
              </h3>
            </div>
          </div>
        </section>
      </div>
    );
  }
}
function mapDispatchToProps(dispatch) {
  return {
    loginData: login => dispatch(loginData(login)),
    signUp: user => dispatch(signUp(user))
  };
}

function mapStateToProps(state) {
  return {
    aaa: state.newLogin,
    newUser: state.newUser
  };
}

const Form = connect(mapStateToProps, mapDispatchToProps)(Auth);
export default Form;
