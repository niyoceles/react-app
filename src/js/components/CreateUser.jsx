// src/js/components/Form.jsx
import React, { Component } from 'react';
import { connect } from 'react-redux';
import uuidv1 from 'uuid';
import { createUser } from '../actions/index';

function mapDispatchToProps(dispatch) {
  return {
    createUser: user => dispatch(createUser(user)),
  };
}

export class CreateUser extends Component {
  constructor() {
    super();
    this.state = {
      firstName: '',
      lastName: '',
      email: '',
      phone: '',
      userName: '',
      password: '',
      location: '',
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(event) {
    this.setState({ [event.target.id]: event.target.value });
  }

  handleSubmit(e) {
    e.preventDefault();

    const createUserData = {
      id: uuidv1(),
      firstName: this.state.firstName,
      lastName: this.state.lastName,
      email: this.state.email,
      phone: this.state.phone,
      userName: this.state.userName,
      password: this.state.password,
      location: this.state.location,
    };
    // call action
    this.props.createUser(createUserData);
  }

  render() {
    const { firstName } = this.state;
    const { lastName } = this.state;
    const { email } = this.state;
    const { phone } = this.state;
    const { userName } = this.state;
    const { password } = this.state;
    const { location } = this.state;

    // const { item } = this.props.newUser;
    return (
      <div>
        <form onSubmit={this.handleSubmit}>
          <div className="form-group">
            <label>First Name</label>
            <input
              type="text"
              className="form-control"
              id="firstName"
              value={firstName}
              onChange={this.handleChange}
            />
          </div>
          <div className="form-group">
            <label>Last Name</label>
            <input
              type="text"
              className="form-control"
              id="lastName"
              value={lastName}
              onChange={this.handleChange}
            />
          </div>
          <div className="form-group">
            <label>Email</label>
            <input
              type="text"
              className="form-control"
              id="email"
              value={email}
              onChange={this.handleChange}
            />
          </div>

          <div className="form-group">
            <label>Phone</label>
            <input
              type="text"
              className="form-control"
              id="phone"
              value={phone}
              onChange={this.handleChange}
            />
          </div>

          <div className="form-group">
            <label>Username</label>
            <input
              type="text"
              className="form-control"
              id="userName"
              value={userName}
              onChange={this.handleChange}
            />
          </div>

          <div className="form-group">
            <label>Password</label>
            <input
              type="password"
              className="form-control"
              id="password"
              value={password}
              onChange={this.handleChange}
            />
          </div>
          <div className="form-group">
            <label>Location</label>
            <input
              type="text"
              className="form-control"
              id="location"
              value={location}
              onChange={this.handleChange}
            />
          </div>

          <button type="submit" className="btn btn-success btn-lg">
          Create User
          </button>
        </form>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    newUser: state.newUser,
  };
}
<<<<<<< HEAD
const CreateUser = connect(mapStateToProps, mapDispatchToProps)(userForm);
export default CreateUser;
=======
const creatingUser = connect(mapStateToProps, mapDispatchToProps)(CreateUser);
export default creatingUser;
>>>>>>> develop
