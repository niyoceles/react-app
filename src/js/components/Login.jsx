// src/js/components/Form.jsx
import React, { Component } from "react";
import { connect } from "react-redux";
import { loginData } from "../actions/index";

function mapDispatchToProps(dispatch) {
  return {
    loginData: login => dispatch(loginData(login))
  };
}

class LoginForm extends Component {
  constructor() {
    super();
    this.state = {
      email: "",
      password: ""
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }
  handleChange(event) {
    this.setState({ [event.target.id]: event.target.value });
  }
  handleSubmit(event) {
    event.preventDefault();
    const postLoginData = {
        email: this.state.email,
        password: this.state.password
      }
    this.props.loginData(postLoginData);
  }
  
  render() {
    const { email } = this.state;
    const { password } = this.state;
    return (
      <form onSubmit={this.handleSubmit}>
        <div className="form-group">
          <label htmlFor="email">email</label>
          <input
            type="text" name="email"
            className="form-control"
            id="email"
            value={email}
            onChange={this.handleChange}
          />
        </div>

        <div className="form-group">
          <label htmlFor="password">password</label>
          <input
            type="text" name="password"
            className="form-control"
            id="password"
            value={password}
            onChange={this.handleChange}
          />
        </div>
        <button type="submit" className="btn btn-success btn-lg">
          Login
        </button>
      </form>
    );
  }
}
const Form = connect(null, mapDispatchToProps)(LoginForm);
export default Form;