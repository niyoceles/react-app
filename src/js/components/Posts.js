import React, { Component } from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';
import { fetchTask } from "../actions/index";

class Posts extends Component {
  
  componentWillMount() {
   this.props.fetchTask();
   console.log(this.props.remoteUsers);
  }

  componentWillReceiveProps(nextProps){
    if(nextProps.new){
    this.props.remoteUsers.unshift(nextProps.newTask);
    }

  }

  render() {
    const PostItem = this.props.remoteUsers.map(post => (
      <div key={post.id}>
        <h3>{post.title}</h3>
        {post.body}
      </div>
    ))
    return (

      <div>
        Posts
        {PostItem}
      </div>
    );
  }
}

Posts.propTypes = {
  fetchTask: PropTypes.func.isRequired,
  Posts: PropTypes.array.isRequired,
};
const mapStateToProps = state =>({
 remoteUsers: state.remoteUsers.items,
});
export default connect(mapStateToProps, {fetchTask})(Posts);