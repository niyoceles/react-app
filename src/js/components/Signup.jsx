/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import uuidv1 from 'uuid';
import { signUp } from '../actions/index';

function mapDispatchToProps(dispatch) {
  return {
    signUp: user => dispatch(signUp(user)),
  };
}

export class SignUp extends Component {
  constructor() {
    super();
    this.state = {
      firstName: '',
      lastName: '',
      email: '',
      phone: '',
      userName: '',
      password: '',
      location: '',
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(event) {
    this.setState({ [event.target.id]: event.target.value });
  }

  handleSubmit(e) {
    e.preventDefault();
    const {
      firstName, lastName, email, phone, userName, password, location,
    } = this.state;

    const signUpData = {
      id: uuidv1(),
      firstName,
      lastName,
      email,
      phone,
      userName,
      password,
      location,
    };
    // call action
    // eslint-disable-next-line react/prop-types
    this.props.signUp(signUpData);
  }

  render() {
    const {
      firstName, lastName, email, phone, userName, password, location,
    } = this.state;

    const styleSignup = {
      border: '4px solid pink',
      width: '50%',
      margin: '0 auto',
    };

    const userSignup = this.props.newUser.item;
    localStorage.setItem('token', userSignup.token);
    localStorage.setItem('userDetails', JSON.stringify(userSignup.data));
    if (userSignup.status === 201) {
      alert('Successful signup');
      window.location = './users';
    }
    console.log(userSignup);

    return (
      <div style={styleSignup}>
        <h3>Sign form</h3>
        <hr />
        <form onSubmit={this.handleSubmit}>
          <div className="form-group">
            <label>First Name</label>
            <input
              type="text"
              className="form-control"
              id="firstName"
              value={firstName}
              onChange={this.handleChange}
            />
          </div>
          <div className="form-group">
            <label>Last Name</label>
            <input
              type="text"
              className="form-control"
              id="lastName"
              value={lastName}
              onChange={this.handleChange}
            />
          </div>
          <div className="form-group">
            <label>Email</label>
            <input
              type="text"
              className="form-control"
              id="email"
              value={email}
              onChange={this.handleChange}
            />
          </div>

          <div className="form-group">
            <label>Phone</label>
            <input
              type="text"
              className="form-control"
              id="phone"
              value={phone}
              onChange={this.handleChange}
            />
          </div>

          <div className="form-group">
            <label>Username</label>
            <input
              type="text"
              className="form-control"
              id="userName"
              value={userName}
              onChange={this.handleChange}
            />
          </div>

          <div className="form-group">
            <label>Password</label>
            <input
              type="password"
              className="form-control"
              id="password"
              value={password}
              onChange={this.handleChange}
            />
          </div>
          <div className="form-group">
            <label>Location</label>
            <input
              type="text"
              className="form-control"
              id="location"
              value={location}
              onChange={this.handleChange}
            />
          </div>

          <button type="submit" className="btn btn-success btn-lg">
          Signup
          </button>
        </form>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    newUser: state.newUser,
  };
}
const creatingUser = connect(mapStateToProps, mapDispatchToProps)(SignUp);
export default creatingUser;
