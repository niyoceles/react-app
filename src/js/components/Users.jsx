import React, { Component } from "react";
import { connect } from "react-redux";
import { getUsers } from "../actions/index";

export class User extends Component {
  constructor() {
    super();
  }
  componentWillMount(){
    this.props.getUsers();
  }

  render() {
    return (
      <ul className="list-group list-group-flush">
        <h2>Users</h2>
        {this.props.remoteUsers.map(user => {
          return (<li className="list-group-item" key={user.id}>
            {user.title}
          </li>);
        })}
      </ul>
    );
  }
}
function mapStateToProps(state) {
  return {
    remoteUsers: state.remoteUsers
  };
}
export default connect(
  mapStateToProps,
  { getUsers }
)(User);