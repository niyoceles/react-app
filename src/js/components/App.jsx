import React, { Component } from 'react';
import {
  BrowserRouter as Router, Route, Switch, Link
} from 'react-router-dom';

import Auth from './Auth.jsx';
import Users from './Users.jsx';
import '../../css/style.css';
import '../../assets/js/app';

class App extends Component {
  render() {
    // const ulStyle = {
    //   width: '350px',
    //   height: '50px',
    // };
    // const linkStyle = {
    //   color: 'white',
    // };
    return (
      <div>
        <Router>
          {/* <ul className="nav nav-pills">
            <button style={ulStyle} className="btn-primary">
              <Link to="/signup" style={linkStyle}>User Sign Up</Link>
            </button>
            <button style={ulStyle} className="btn-primary btn-md">
              <Link to="/" style={linkStyle}>User Sign In</Link>
            </button>
            <button style={ulStyle} className="btn-primary ">
              <Link to="/users" style={linkStyle}>Users</Link>
            </button>
          </ul>
          <hr /> */}
          <Switch>
            <Route exact path="/" component={Auth} />
            <Route path="/users" component={Users} />
          </Switch>
        </Router>
      </div>

    );
  }
}

export default App;
