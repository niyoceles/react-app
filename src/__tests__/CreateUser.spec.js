import React from 'react';
import { shallow } from 'enzyme';
import { CreateUser } from '../js/components/CreateUser';


describe('testing Create user component', () => {
  test('should render create user component', () => {
    const props = { createUser: jest.fn() };
    const wrapper = shallow(<CreateUser {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
