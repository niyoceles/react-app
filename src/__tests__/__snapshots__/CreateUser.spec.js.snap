// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Create user component should render create user component 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label>
        First Name
      </label>
      <input
        className="form-control"
        id="firstName"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        Last Name
      </label>
      <input
        className="form-control"
        id="lastName"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        Email
      </label>
      <input
        className="form-control"
        id="email"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        Phone
      </label>
      <input
        className="form-control"
        id="phone"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        Username
      </label>
      <input
        className="form-control"
        id="userName"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        Password
      </label>
      <input
        className="form-control"
        id="password"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        Location
      </label>
      <input
        className="form-control"
        id="location"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <button
      className="btn btn-success btn-lg"
      type="submit"
    >
      Create User
    </button>
  </form>
</div>
`;
